### PR TITLE
Use environment variables only for service config

### PR DESCRIPTION
**Pull request by kenoir**
Passing config vars to services not looks like:

```hcl
module "my_service" {
   // ...  
  config_vars = [
    {
      name  = "SOME_VAR"
      value = "${var.some_var}"
    },
    {
      name  = "ANOTHER_VAR"
      value = "${var.another_var}"
    }
  ]
  // ...
}
```

### CODE REVIEW

File: service/ecs_task/data.tf [hunk 1]
```diff
@@ -12,7 +12,7 @@ data "template_file" "definition" {
     secondary_container_port = "${var.secondary_container_port}"
     volume_name              = "${var.volume_name}"
     container_path           = "${var.container_path}"
-    environment_vars         = "[${join(",", concat(var.service_vars,var.extra_vars))}]"
+    environment_vars         = "[${jsonencode(concat(var.service_vars,var.config_vars))}]"
```
Comment (jamesgorrie) [inline service/ecs_task/data.tf]: 💯

File: service/ecs_task/variables.tf [hunk 2]
```diff
@@ -51,7 +51,7 @@ variable "service_vars" {
   type        = "list"
 }
 
-variable "extra_vars" {
+variable "config_vars" {
```
Comment (jamesgorrie) [inline service/ecs_task/variables.tf]: Why not call them env vars?

File: service/README.md [hunk 3]
```diff
@@ -17,9 +17,12 @@ module "my_service" {
...
 
-  config_vars = {
-    my_config_var = "${var.my_config_var}"
-  }
+  config_vars = [
+    {
+      name  = "MY_VAR"
+      value = "${var.my_config_var}"
+    }
+  ]
```
Comment (alexwlchan) [inline service/README.md]: needs changing

File: sqs_autoscaling_service/main.tf [hunk 4]
```diff
@@ -20,12 +20,8 @@ data "aws_ecs_cluster" "cluster" {
   cluster_name = "${var.cluster_name}"
 }
 
-locals {
-  config_template = "${var.config_template == "" ? var.name : var.config_template}"
-}
-
 module "service" {
-  source = "git::https://github.com/wellcometrust/terraform-modules.git//service?ref=v3.0.0"
+  source = "git::https://github.com/wellcometrust/terraform-modules.git//service?ref=nicer-service-vars"
```
Comment (alexwlchan) [inline sqs_autoscaling_service/main.tf]: needs changing to v4.0.0

**Comment by alexwlchan** [comment]
@jamesgorrie You may be interested in the extra commits I added – I was able to bend HCL to my will so the interface for config variables doesn’t have to change; the name/value abstraction is absorbed by the module.

That aside, this is ready to go (and tested on our ID minter). Merging! 🎉

**Comment by jamesgorrie** [comment]
@alexwlchan any reason nt to be opinionated about it?

**Comment by alexwlchan** [comment]
> any reason n0t to be opinionated about it?

I don’t understand the question.

**Pull request merged**